### ISSUE TITLE
Developer docs

### ISSUE THREAD

**Issue by mbillingr** [scot-dev/scot#127]
As discussed by email, let's put our workflows in the Github Wiki:

- [ ] Working with git :smile:
- [ ] Building the docs
- [ ] Making a release

**Comment by cle1109** [comment]
*Ahem*, in the meantime I worked through a very nice tutorial by Atlassian. I think I do understand what I'm doing now provided that I'm using the command line :-). Anyway, it's a good idea to document that we're using a forking workflow and the basic commands required to implement it. Which means that we should really create PRs only from our forked repos and never directly push to scot-dev master. Except maybe for important hotfixes. What's your take on that?

**Comment by mbillingr** [comment]
> create PRs only from our forked repos and never directly push to scot-dev master.

Agreed.

> Except maybe for important hotfixes.

I think even those should go through a PR.

> nice tutorial by Atlassian

I'd like to read that too. Can you send me the link? Currently, my git knowledge is all Stack Overflow :)

**Comment by cle1109** [comment]
OK great, then let's try to stick to a strict forking workflow in the future. We have one open PR #62 which is a branch from this repo. How do we proceed with this one? All other PRs are already from our own repos.

I've added a link to the tutorial in my original comment above.

**Comment by mbillingr** [comment]
> How do we proceed with this one?

I moved the branch to my fork and created a new PR.

**Comment by cle1109** [comment]
Thanks! Now there's only the `icatests` branch - what do we do with this one? It doesn't seem to be associated with any PR.

**Comment by mbillingr** [comment]
I can't even remember what it was used for. Are there any commits in it?

**Comment by cle1109** [comment]
Yes: https://github.com/scot-dev/scot/compare/icatests

Regarding creating a wiki, I just saw that the Pandas folks are putting developer information into a file:
https://github.com/pydata/pandas/blob/master/CONTRIBUTING.md

Maybe this is better than a Wiki?

**Comment by mbillingr** [comment]
I remember now. The `icatests` branch was created when we wanted to compare different ICA implementations after discussion with someone. We could move this into a gist and dispose the branch.

> Maybe [a file] is better than a Wiki?

I don't know. When putting it in a file we'd have to go through all the PR procedure whenever we change something. Is this desirable?

**Comment by cle1109** [comment]
Good idea, please go ahead and move the relevant file to a gist. What about the changes in `topoplot.py`?

I'd prefer the file - I like it if information lives in a simple accessible format. Who knows how difficult it is to get that out of a wiki?

**Comment by mbillingr** [comment]
I'll look into the branch.

That's a good argument for the file. Would be nice if we could somehow link to the file, because unlike README.md it is not shown automatically.

**Comment by cle1109** [comment]
Somehow it's possible to show a link to this file when submitting a new issue:
https://github.com/pydata/pandas/issues/new

We should set this up as well. We could also refer to `CONTRIBUTING.md` in our `README.md`.

**Comment by mbillingr** [comment]
Nice. This happens [automagically](https://github.com/blog/1184-contributing-guidelines)
